frontend/server/src/main/java/com/amazonaws/ml/mms/util/ConfigManager.java

```diff
@@ -100,8 +106,43 @@ public boolean isDebug() {
                 || Boolean.parseBoolean(prop.getProperty(DEBUG, "false"));
     }
 
-    public int getPort() {
-        return getIntProperty(PORT, isUseSsl() ? 8443 : 8080);
+    public URI getAddressProperty(String key, String host, int port)
+            throws IllegalArgumentException {
+        String address = getProperty(key, "");
+        IllegalArgumentException badAddress =
```

Comment:
1. create exception only where you needed
2. String.format is slow, just use "Invalid address: " + address